docs/2.guide/4.recipes/4.sessions-and-authentication.md

```diff
@@ -0,0 +1,403 @@
+
+You'll need a `users` table in your database with the following columns:
+- `id` (int, primary key, auto increment)
+- `email` (varchar)
+- `password` (varchar)
+
+Additionally, we'll use [nuxt-aut-utils](https://github.com/Atinux/nuxt-auth-utils) by [Atinux](https://github.com/Atinux) to handle the authentication and session management.
+
+## Steps
+
+### 1. Install nuxt-auth-utils
+
+
+Install the [auth-utils](https://github.com/Atinux/nuxt-auth-utils) module using the `nuxi` CLI.
+
+
+```bash
+npx nuxi@latest module add auth-utils
+```
+
+### 1a. (Optional) Add a session encryption key
+
+Session cookies are encrypted. The encryption key is set from the `.env` file. This key will be added to your `.env` automatically when running in development mode the first time. However, you'll need to add this to your production environment before deploying.
+
+```dotenv [.env]
+NUXT_SESSION_PASSWORD=password-with-at-least-32-characters
+```
+
+### 2. Create a registration page
+
+The first page we'll need is a page for users to register and create new accounts. Create a new Vue page in your Nuxt app at `/pages/register.vue` for user registration. This page should have a form with fields for email and password. We'll intercept the form submission using `@submit.prevent` and use the [`$fetch`](/docs/getting-started/data-fetching#fetch) utility to post the data to `/api/register`. This form POST will be received by Nuxt in an API route which we will set up next. 
+
+If the request is successful, we'll navigate to the (soon to be created) `/users` page, which will be guarded and only be visible to logged in users.
+
+Here's an example registration form for reference:
+
+```vue [pages/register.vue]
+<script setup lang="ts">
+  const form = ref({
+    name: "",
+    email: "",
+    password: "",
+  });
+
+  const error = ref(null);
+
+  async function submitForm() {
+    // clear any previous errors
+    error.value = null;
+
+    // perform the login
+    try {
+      await $fetch("/api/register", {
+        method: "POST",
+        body: form.value,
+      });
+    } catch (e) {
+      // if there's an error, set the error message and return early
+      error.value = "Error Registering User";
+      return;
+    }
+    // Refresh the session status now that the user is logged in
+    // This is a fetch() function from auth-utils to refresh session data in the client, and is not the same as $fetch
+    const { fetch } = useUserSession();
+    await fetch();
+    // you may want to use something like Pinia to manage global state of the logged-in user
+    // update Pinia state here...
+
+    // take the user to the auth-only users index page now that they're logged in
```

Comment:
"the auth-only users index page" sounds odd. Perhaps it can be rephrased into "the authorized users page"?